aikau/src/main/resources/alfresco/html/Image.js

```diff
@@ -0,0 +1,421 @@
+          */
+         resize: function alfresco_html_Image__resize(forceNatural) {
+
+            // Use configured width/height first
+            if ((this.width || this.height) && !forceNatural) {
+
+               // Just style according to whichever dimension(s) we have
+               domStyle.set(this.imageNode, {
+                  width: this.width || "auto",
+                  height: this.height || "auto"
+               })
+
+            } else {
+
+               // Are there any CSS dimensions we should leave in-place
+               var cssDimensions = this.getCssDimensions(),
+                  hasCssDimensions = cssDimensions.w || cssDimensions.h;
+
+               // Do the resize only if no CSS dimensions, or if forceNatural is true
+               if (!hasCssDimensions || forceNatural) {
+                  this._updateNaturalDimensions().then(lang.hitch(this, function() {
+
+                     // Use the natural dimensions, unless neither exist
+                     var w = this.naturalWidth,
+                        h = this.naturalHeight;
+                     if (!w && !h) {
+                        w = h = this.defaultMinimumSize;
+                     }
+
+                     // Update the dimensions
+                     domStyle.set(this.imageNode, {
+                        width: w + "px",
+                        height: h + "px"
+                     });
+                  }));
+               }
+            }
+         },
+
+         /**
+          * Get the URL of the background image, which is assumed to be a single image,
+          * not surrounded by quotes (which would break in IE).
+          *
+          * @instance
+          * @returns {string} The URL of the background image, or null if not available
+          */
+         _getBackgroundImageUrl: function alfresco_html_Image___getBackgroundImageUrl() {
+
+            // Setup variables
+            var url = null,
+               urlRegex = /url\((.+)\)/,
+               imageStyle = this.imageNode.currentStyle || getComputedStyle(this.imageNode), // IE8 || Proper Browser
+               backgroundImage = imageStyle.backgroundImage,
+               regexMatch;
+
+            // If we have a background image property, extract the URL
+            if (backgroundImage) {
+               regexMatch = urlRegex.exec(backgroundImage);
+               if (regexMatch) {
+                  url = regexMatch[1];
+               }
+            }
+
+            // Pass back the URL
+            return url;
+         },
+
+         /**
+          * Get the natural size of the currently displayed image.
+          *
+          * @instance
+          * @param {boolean} [force] Whether to force retrieval if already present
+          * @returns {external:dojo/promise/Promise} A promise that will resolve once dimensions are retrieved
+          */
+         _updateNaturalDimensions: function alfresco_html_Image___updateNaturalDimensions(force) {
```

Comment:
Broken, I think, in the sense that background images will not display because the natural width/height is needed to make the blank.gif image big enough to show them.